rest/route/agent.go

```diff
@@ -789,6 +790,45 @@ func (h *attachTestLogHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(logReply)
 }
 
+// POST /task/{task_id}/test_results
+type attachTestResultsHandler struct {
+	settings *evergreen.Settings
+	results  []testresult.TestResult
+	taskID   string
+}
+
+func makeAttachTestResults(settings *evergreen.Settings) gimlet.RouteHandler {
+	return &attachTestResultsHandler{
+		settings: settings,
+	}
+}
+
+func (h *attachTestResultsHandler) Factory() gimlet.RouteHandler {
+	return &attachTestResultsHandler{
+		settings: h.settings,
+	}
+}
+
+func (h *attachTestResultsHandler) Parse(ctx context.Context, r *http.Request) error {
+	if h.taskID = gimlet.GetVars(r)["task_id"]; h.taskID == "" {
+		return errors.New("missing task ID")
+	}
+	err := utility.ReadJSON(r.Body, &h.results)
+	if err != nil {
+		return errors.Wrap(err, "reading test results from JSON request body")
+	}
+	return nil
+}
+
+func (h *attachTestResultsHandler) Run(ctx context.Context) gimlet.Responder {
+	// TODO: DEVPROD-16200 Implement the new DB/S3-backed Evergreen test results service
+	grip.Debug(message.Fields{
+		"message": "received test results",
+		"results": h.results,
```

Comment:
i'm assuming this is just in while you're doing initial testing, but we should remove it before using this route long term so we don't spam splunk

Reply:
+1 this could be a lot of data if we leave it in.